cpp/models/abm/README.md

```diff
@@ -1,57 +1,127 @@
 # Agent Based Model
 
-This module models and simulates the epidemic using an agent based model (*ABM*) approach. Unlike the SECIR compartment model that uses an ODE, this model simulates the spread of COVID19 in a population with actual persons (the agents) moving throughout the world and interacting with each other.
+This module models and simulates the epidemic using an agent based model (*ABM*) approach. Unlike the SECIR compartment model that uses an ODE, this model simulates the spread of COVID19 in a population with actual persons (the agents) moving throughout locations in the world and interacting (infecting) with each other.
 
 ## Structure
 
 The model consists of the following major classes:
-1. Person: represents the agents of the model. A person has an assigned infection state (similar to the compartments of the SECIR model) and location. 
-2. Location: represents locations in the world where people meet and interact, e.g. home, school, work, social events.
-3. World: collection of all persons and locations.
-4. Simulation: run the simulation and store results.
+
+1. Person: represents the agents of the model. A person has an ID, i.e. a unique number, an age, a location and a list with their assigned locations, i.e. the locations they visit during the simulation. They can perform tests and wear masks. Every person has lists with past and current infections and vaccinations.
+2. Infection: collection of all information about a persons infection, i.e. infectiousness, infection course, virus variant. The infection course is drawn stochastically from the infection states that are similar to the compartments of the SECIR model.
+3. Location: represents locations in the world where people meet and interact, e.g. home, school, work, social events. A location can be split into cells to model parts of a location like classrooms in a school. Some infection parameters are location specific and one can activate NPIs like mandatory masks or tests to enter the location.
```

Comment:
- 'a location represents a location' is tautological, I suggest using 'place' instead
- 'social events' are not a place. I suggest 'social event sites'